src/Cli/dotnet/commands/dotnet-workload/install/NetSdkMsiInstallerClient.cs

```diff
@@ -476,6 +482,91 @@ public void Shutdown()
             _shutdown = true;
         }
 
+        public PackageId GetManifestPackageId(ManifestId manifestId, SdkFeatureBand featureBand)
+        {
+            return new PackageId($"{manifestId}.Manifest-{featureBand}.Msi.{RuntimeInformation.ProcessArchitecture.ToString().ToLowerInvariant()}");
+        }
+
+        public async Task ExtractManifestAsync(string nupkgPath, string targetPath)
+        {
+            string extractionPath = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+            if (Directory.Exists(extractionPath))
+            {
+                Directory.Delete(extractionPath, true);
+            }
+
+            try
+            {
+                Directory.CreateDirectory(extractionPath);
+
+                Log?.LogMessage($"Extracting '{nupkgPath}' to '{extractionPath}'");
+                await _nugetPackageDownloader.ExtractPackageAsync(nupkgPath, new DirectoryPath(extractionPath));
+                if (Directory.Exists(targetPath))
+                {
+                    Directory.Delete(targetPath, true);
+                }
+
+                //  Do we need to worry about signing validation here?  I think not, as if the manifest is ever actually installed, it will go through signing validation
+                string extractedManifestPath = Path.Combine(extractionPath, "data", "extractedManifest");
+                if (Directory.Exists(extractedManifestPath))
+                {
+                    Log?.LogMessage($"Copying manifest from '{extractionPath}' to '{targetPath}'");
+                    Directory.CreateDirectory(Path.GetDirectoryName(targetPath));
+                    FileAccessRetrier.RetryOnMoveAccessFailure(() => DirectoryPath.MoveDirectory(extractedManifestPath, targetPath));
+                }
+                else
+                {
+                    string packageDataPath = Path.Combine(extractionPath, "data");
+                    if (!Cache.TryGetMsiPathFromPackageData(packageDataPath, out string msiPath, out _))
+                    {
+                        throw new FileNotFoundException("Manifest MSI not found in NuGet package", extractionPath);
+                    }
+                    string msiExtractionPath = Path.Combine(extractionPath, "msi");
+                    
+                    var psi = new ProcessStartInfo()
+                    {
+                        FileName = "msiexec",
+                        ArgumentList =
+                        {
+                            "/a",
+                            msiPath,
+                            $"TARGETDIR={msiExtractionPath}",
+                            "/QN",  //  Quiet: No UI
+                        }
+                    };
```

Comment:
You should be able to get rid of this and use the native APIs we have in the CLI. `WindowsInstaller.InstallProduct` allows you to do that. The doc comments actually mentions admin installs.